CMakeLists.txt

```diff
@@ -40,51 +39,54 @@ target_include_directories(${PROJECT_NAME} INTERFACE
 ##
 # Unit Tests
 ##
-set(TEST_SOURCE_FILES
-    ${SG14_TEST_SOURCE_DIRECTORY}/main.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/flat_map_test.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/flat_set_test.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/inplace_function_test.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/ring_test.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/slot_map_test.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/uninitialized_test.cpp
-    ${SG14_TEST_SOURCE_DIRECTORY}/unstable_remove_test.cpp
-)
-
-set(TEST_NAME ${PROJECT_NAME}_tests)
-add_executable(${TEST_NAME} ${TEST_SOURCE_FILES})
-target_link_libraries(${TEST_NAME} ${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT})
-target_include_directories(${TEST_NAME} PRIVATE "${SG14_TEST_SOURCE_DIRECTORY}")
-
-# Compile options
-if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-	target_compile_options(${TEST_NAME} PRIVATE -Wall -Wextra -Werror)
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-	target_compile_options(${TEST_NAME} PRIVATE -Wall -Wextra -Werror)
-	set_source_files_properties(${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp PROPERTIES
-		COMPILE_FLAGS "-Wno-unused-parameter"
+if (SG14_ENABLE_UNIT_TESTS)
+	set(SG14_TEST_SOURCE_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/SG14_test")
+
+	set(TEST_SOURCE_FILES
+		${SG14_TEST_SOURCE_DIRECTORY}/main.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/flat_map_test.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/flat_set_test.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/inplace_function_test.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/ring_test.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/slot_map_test.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/uninitialized_test.cpp
+		${SG14_TEST_SOURCE_DIRECTORY}/unstable_remove_test.cpp
 	)
+
+	set(TEST_NAME ${PROJECT_NAME}_tests)
+	add_executable(${TEST_NAME} ${TEST_SOURCE_FILES})
+	target_link_libraries(${TEST_NAME} ${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT})
+	target_include_directories(${TEST_NAME} PRIVATE "${SG14_TEST_SOURCE_DIRECTORY}")
+
+	# Compile options
+	if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+		target_compile_options(${TEST_NAME} PRIVATE -Wall -Wextra -Werror)
+	elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+		target_compile_options(${TEST_NAME} PRIVATE -Wall -Wextra -Werror)
+		set_source_files_properties(${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp PROPERTIES
+			COMPILE_FLAGS "-Wno-unused-parameter")
```

Comment:
Could you uncuddle this parenthesis again, so that `git show -b` will have one fewer gratuitous diff?

Reply:
Done